prov/efa/src/rxr/rxr_pkt_type_data.c

```diff
@@ -229,12 +229,21 @@ ssize_t rxr_pkt_send_data_desc(struct rxr_ep *ep,
 void rxr_pkt_handle_data_send_completion(struct rxr_ep *ep,
 					 struct rxr_pkt_entry *pkt_entry)
 {
+	struct rxr_data_pkt *data_pkt;
 	struct rxr_tx_entry *tx_entry;
 
-	tx_entry = (struct rxr_tx_entry *)pkt_entry->x_entry;
-	tx_entry->bytes_acked +=
-		rxr_get_data_pkt(pkt_entry->pkt)->hdr.seg_size;
+	data_pkt = rxr_get_data_pkt(pkt_entry->pkt);
+	if (data_pkt->hdr.flags != 0)
```

Comment:
You can dereference the `x_entry` field of pkt_entry, obtain the tx_entry, and look at `tx_entry->flags` instead of adding this to the data packet header.

The deref is already happening on line 245 so all you need to do is change this if block and move it.